datumaro/components/operations.py

```diff
@@ -825,15 +832,7 @@ class _ShapeMerger(AnnotationMerger, _ShapeMatcher):
     def merge_clusters(self, clusters):
         merged = []
         for cluster in clusters:
-            label, label_score = self.find_cluster_label(cluster)
-            shape, shape_score = self.merge_cluster_shape(cluster)
-
-            shape.z_order = max(cluster, key=lambda a: a.z_order).z_order
-            shape.label = label
-            shape.attributes['score'] = label_score * shape_score \
-                if label is not None else shape_score
-
-            merged.append(shape)
+            merged.append(self.merge_cluster(cluster))
```

Comment:
Seems like this whole function could now be reduced to `return list(map(self.merge_cluster, clusters))`.

Reply:
Done